### PR TITLE
update build to create merged firmware file

### PR DESCRIPTION
**Pull request by sfranzyshen**
update build to create merged firmware file for use with esp-web-tools webserial based firmware installer
... next up support for multiple (s2, s3, and c3) builds ...

### CODE REVIEW

File: scripts/build-mp.sh [hunk 1]
```diff
@@ -40,6 +40,4 @@ WD=$(pwd)
  cp build-WROVER_16M-OTA_USER/partition_table/partition-table.bin "${WD}/images" &&
  cp build-WROVER_16M-OTA_USER/bootloader/bootloader.bin "${WD}/images" &&
  cp build-WROVER_16M-OTA_USER/ota_data_initial.bin "${WD}/images" && cd "${WD}" &&
```
Comment (mishafarms) [inline scripts/build-mp.sh]: I don't think the ending && cd "${WD}" && should be needed. The "(" ")" mean do all this in a sub-command and we should be back in the WD when this is all done.

Did you see some other behaviour?

File: scripts/build-merge.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+ PART_ADDR=$(./gen_esp32part.py images/partition-table.bin | grep vfs | awk -F',' '{print $4}')
+ esptool.py --chip esp32 merge_bin -o images/merged-firmware.bin --flash_mode dio --flash_size 16MB --flash_freq 40m 0x1000 images/bootloader.bin 0x8000 images/partition-table.bin 0x10000 images/micropython.bin 0x510000 images/ota_data_initial.bin "${PART_ADDR}" images/sys.img
```
Comment (mishafarms) [inline scripts/build-merge.sh]: For the partitions that are movable.
ota_0,app,ota_0,0x10000,2560K,
otadata,data,ota,0x510000,8K,
I would prefer to get them from the partition table like the VFS. Also we can set some variables for the ones that are
pre-defined.
BOOT_ADDR=0x1000
PART_ADDR=0x8000
OTA_ADDR=$(.gen_esp32part.py images/partition-table.bin | grep ota_0 | awk -F',' '{print $4}')
OTA_DATA_ADDR=$(.gen_esp32part.py images/partition-table.bin | grep otadata | awk -F',' '{print $4}')
Then the esptool.py command can look like this.
esptool.py --chip esp32 merge_bin -o images/merged-firmware.bin --flash_mode dio --flash_size 16MB --flash_freq 40m "${BOOT_ADDR}" images/bootloader.bin "${PART_ADDR}" images/partition-table.bin "${OTA_ADDR}" images/micropython.bin "${OTA_DATA_ADDR}" images/ota_data_initial.bin "${PART_ADDR}" images/sys.img

**Comment by mishafarms** [comment]
You need to set the permissions to execute for all (chmod a+x) on your new script files and check them in.

File: scripts/build-mp.sh (hunk 1 above)
Comment (sfranzyshen) [reply scripts/build-mp.sh]: sorry  should have tested everything before I opened the pull request ... but  made changes and went threw it ... reflected n the changes in the pull

File: scripts/build-merge.sh (hunk 2 above)
Comment (sfranzyshen) [reply scripts/build-merge.sh]: I agree movable partitions ...  I was going to address that with the next round of changes to handle multiple builds

**Comment by sfranzyshen** [comment]
> You need to set the permissions to execute for all (chmod a+x) on your new script files and check them in.

 added these thru the web ..  I will clone the archive with github-desktop and fix them

File: README.md [hunk 3]
```diff
@@ -26,32 +26,59 @@ For Linux (the default):
...
+
+## Build panel-web-app
+
+	cd ~/src/esp32-micropython/panel
+	yarn
+	yarn run build
+
+## Build EduBlocks
+
+	cd ~/src/edublocks-micropython
```
Comment (mishafarms) [inline README.md:48]: This should probably have the mount-sys-linux/umount-sys-linux as well

File: scripts/build-merge.sh (hunk 2 above)
Comment (mishafarms) [reply scripts/build-merge.sh]: I am OK with that.

**Pull request merged**